astropy/io/misc/tests/test_yaml.py

```diff
@@ -293,3 +293,22 @@ def test_ecsv_astropy_objects_in_meta():
     compare_time(tm, t2.meta["tm"])
     compare_coord(c, t2.meta["c"])
     assert t2.meta["unit"] == unit
+
+
+def test_yaml_dump_of_object_arrays_fail():
+    """Test that dumping and loading object arrays fails."""
+    with pytest.raises(TypeError, match="Cannot serialize"):
+        dump(np.array([1, 2, 3], dtype=object))
+
+
+def test_yaml_load_of_object_arrays_fail():
+    """Test that dumping and loading object arrays fails."""
```

Comment:
Can you add text about how you made the test example in this test? Presumably `dump(np.array([1, 2, 3], dtype=object))` before putting in the exception for that.

Reply:
Hi @taldcroft, it's not very clear to me what you are asking. Basically, to select the test case, I dumped the array above (`np.array([1, 2, 3], dtype=object)`) to a file, closed ipython to lose the reference, reopened, and loaded the file. I got the same segmentation fault of the issue report. What should I write here?

Reply:
Forget my previous comment, I now get what you were saying and (I hope) addressed the comment